repos/system_upgrade/el7toel8/actors/checkchrony/actor.py

```diff
@@ -0,0 +1,53 @@
+from leapp.actors import Actor
+from leapp.models import InstalledRedHatSignedRPM
+from leapp.reporting import Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.libraries.common.reporting import report_generic
+from leapp.libraries.stdlib import CalledProcessError, run
+
+
+class CheckChrony(Actor):
+    """
+    Check for incompatible changes in chrony configuration.
+
+    Warn that the default chrony configuration in RHEL8 uses the leapsectz
+    directive.
+    """
+
+    name = 'check_chrony'
+    consumes = (InstalledRedHatSignedRPM,)
+    produces = (Report)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        installed_packages = set()
+
+        signed_rpms = self.consume(InstalledRedHatSignedRPM)
+        for rpm_pkgs in signed_rpms:
+            for pkg in rpm_pkgs.items:
+                installed_packages.add(pkg.name)
+
+        if 'chrony' not in installed_packages:
+            report_generic(
+                    title='chrony package is not installed',
+                    summary='chrony package was not found in the rpm database',
+                    severity='low')
+            return
+
+        # Check if using default chrony.conf
+        try:
+            run(['/bin/sh', '-c', 'rpm -V chrony | grep chrony.conf'])
+            default_chrony_conf = False
+        except (OSError, CalledProcessError):
+            default_chrony_conf = True
```

Comment:
Worth logging an error here in case of OSError, since a OSError shouldn't happen here.

Also I am wondering if it wouldn't be better to do something along these lines:

```python
default_chrony_conf = False
try:
    result = run(['rpm', '-V', 'chrony'], split=False)
    default_chrony_conf = 'chrony.conf' in result['stdout']
except (OSError, CalledProcessError) as e:
    self.log.warn("Failed to call chrony rpm verification failed: %s", str(e))
``` 
**Note** not checked or tested, just an idea

The reason for this is, that I am not a big fan of shell execution like this, and I think it's good enough here to check the output with python, using grep just complicates finding the source of programming errors.

Reply:
This would be nicer, but there is an issue that rpm -V returns with a non-zero exit code when the config files are modified and the run() method doesn't seem to have an option to ignore that. Should I try to extend that function first?

Reply:
This should be fixed now.